services/src/datasets/listing.rs

```diff
@@ -72,24 +71,50 @@ pub enum OrderBy {
 
 /// Listing of stored datasets
 #[async_trait]
-pub trait DatasetProvider: Send
+pub trait DatasetProvider<S: Session>: Send
     + Sync
     + MetaDataProvider<MockDatasetDataSourceLoadingInfo, VectorResultDescriptor, VectorQueryRectangle>
     + MetaDataProvider<OgrSourceDataset, VectorResultDescriptor, VectorQueryRectangle>
     + MetaDataProvider<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle>
-    + ProvenanceProvider
 {
     // TODO: filter, paging
     async fn list(
         &self,
-        // session: &S, // TODO: authorization
+        session: &S,
         options: Validated<DatasetListOptions>,
     ) -> Result<Vec<DatasetListing>>;
 
-    // TODO: is this method useful?
-    async fn load(
-        &self,
-        // session: &S, // TODO: authorization
-        dataset: &DatasetId,
-    ) -> Result<Dataset>;
+    async fn load(&self, session: &S, dataset: &DatasetId) -> Result<Dataset>;
+
+    async fn provenance(&self, session: &S, dataset: &DatasetId) -> Result<ProvenanceOutput>;
+}
+
+#[async_trait]
+pub trait ExternalDatasetProvider: Send
+    + Sync
+    + MetaDataProvider<MockDatasetDataSourceLoadingInfo, VectorResultDescriptor, VectorQueryRectangle>
+    + MetaDataProvider<OgrSourceDataset, VectorResultDescriptor, VectorQueryRectangle>
+    + MetaDataProvider<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle>
+{
+    // TODO: authorization, filter, paging
+    async fn list(&self, options: Validated<DatasetListOptions>) -> Result<Vec<DatasetListing>>;
+
+    // TODO: authorization
+    // TODOis this method useful?
+    async fn load(&self, dataset: &DatasetId) -> Result<Dataset>;
```

Comment:
Why isn't this method useful? If you wouldn't use it, you wouldn't have defined it?

Reply:
it is never used and not even implemented for the existing providers. For internal datasets we use it though and before this PR they had the same trait. That's why it still exists.

Reply:
I guess if we don't use it and do not even implement it, then we should remove it.

Reply:
acac67181c8096c46aa941c5b43072a8f5ee32c9